web/react/components/post_list.jsx

```diff
@@ -55,44 +57,23 @@ module.exports = React.createClass({
             $('.team__header').addClass('theme--gray');
         }
 
+        // Start our Store listeners
         PostStore.addChangeListener(this._onChange);
         ChannelStore.addChangeListener(this._onChange);
         UserStore.addStatusesChangeListener(this._onTimeChange);
         SocketStore.addChangeListener(this._onSocketChange);
 
-        $(".post-list-holder-by-time").perfectScrollbar();
-
-        this.resize();
-
-        var post_holder = $(".post-list-holder-by-time")[0];
-        this.scrollPosition = $(post_holder).scrollTop() + $(post_holder).innerHeight();
-        this.oldScrollHeight = post_holder.scrollHeight;
-        this.oldZoom = (window.outerWidth - 8) / window.innerWidth;
-
-        var self = this;
-        $(window).resize(function(){
-            $(post_holder).perfectScrollbar('update');
-
-            // this only kind of works, detecting zoom in browsers is a nightmare
-            var newZoom = (window.outerWidth - 8) / window.innerWidth;
-
-            if (self.scrollPosition >= post_holder.scrollHeight || (self.oldScrollHeight != post_holder.scrollHeight && self.scrollPosition >= self.oldScrollHeight) || self.oldZoom != newZoom) self.resize();
+        // Timeout exists for the DOM to fully render before making changes
+        var self = this
+        setTimeout(function initialScroll() {
```

Comment:
This seems hacky. A quick google reveals that "DOMContentLoaded" or a JQuery alternative my be a possibility. Why not those?